src/main/java/mate/academy/spring/service/DaoUserDetailsService.java

```diff
@@ -0,0 +1,43 @@
+package mate.academy.spring.service;
+
+import java.util.Optional;
+import java.util.Set;
+import mate.academy.spring.model.Role;
+import mate.academy.spring.model.User;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.stereotype.Service;
+
+@Service
+public class DaoUserDetailsService implements UserDetailsService {
```

Comment:
i think this name is not the best choice. It would be better to replace `Dao` with `Custom` or call it like regular service implementations